UnityProject/Assets/Scripts/Shuttles/MatrixMove.cs

```diff
@@ -141,9 +141,11 @@ public class MatrixMove : ManagedBehaviour
 	/// be rotated to match the target?
 	/// </summary>
 	private bool NeedsRotationClient =>
-		Quaternion.Angle(transform.rotation, InitialFacing.OffsetTo(clientState.FacingDirection).Quaternion) != 0;
+		Quaternion.Angle(TransformRotation, InitialFacing.OffsetTo(clientState.FacingDirection).Quaternion) != 0;
```

Comment:
Whats this change for?

Reply:
so while bounds can be calculate it on other threads